completions/ip

```diff
@@ -8,11 +8,21 @@ _comp_cmd_ip__iproute2_etc()
 
 _comp_cmd_ip__netns()
 {
-    _comp_compgen_split -- "$(
+    _comp_compgen_split -l -- "$(
         {
             ${1-ip} -c=never netns list 2>/dev/null || ${1-ip} netns list
-        } | _comp_awk '{print $1}'
+        } | command sed -e 's/ (.*//g'
     )"
+    # namespace names can have spaces, so we quote all of them if needed
+    local unquoted=("${COMPREPLY[@]}")
+    COMPREPLY=()
+    for i in "${!unquoted[@]}"; do
+        local REPLY
+        printf -v REPLY %q "${unquoted[i]}"
+        _comp_quote "$REPLY"
+        _comp_compgen -a -- -W "$REPLY"
```

Comment:
When using `compgen -W`, one can (and needs to) quote the words inside the option argument as `'"$REPLY"'` because the option argument to `-W` is going to be processed by shell expansions.

```suggestion
        printf -v REPLY %q "${unquoted[i]}"
        _comp_compgen -a -- -W '"$REPLY"'
```

----

In addition, calling `_comp_compgen` for each completion would be slow. We can first collect the results in another array `quoted` and finally perform `((${#quoted[@]})) && _comp_compgen -- -W '"${quoted[@]}"'`.

Reply:
Done